crates/mun_codegen/src/ir/module.rs

```diff
@@ -104,6 +119,19 @@ pub(crate) fn ir_query(db: &impl IrDatabase, file_id: FileId) -> Arc<ModuleIR> {
     let dispatch_table = dispatch_table_builder.finalize(&functions);
     let fn_pass_manager = function::create_pass_manager(&llvm_module, db.optimization_lvl());
 
+    // Create the allocator handle global value
+    let allocator_handle_global = if dispatch_table.has_intrinsic(&crate::intrinsics::new) {
```

Comment:
If `new` is there, `clone` must also be there - and vice versa.

Reply:
Nonetheless if that is not the case in the future this will silently fail. Best to add it just in case.

Reply:
The `new`, `clone`, and - in the future - `drop` are all object management functions. They are all required if a struct type is defined. That should be the real test, I case because checking for everything potential future object management function doesn't scale too well.

Reply:
I agree but this check is to determine whether or not a global variable allocator handle is required. This is only the case if either of these methods is required no matter where they are used. Maybe in the future we would like to use one of these functions for a different use case. Than we dont have to touch the check.

The best check would be to check if one of the functions in the dispatch table are actually using the AllocatorHandle.